server/src/testflinger/api/v1.py

```diff
@@ -988,3 +988,80 @@ def retrieve_token():
     secret_key = os.environ.get("JWT_SIGNING_KEY")
     token = generate_token(allowed_resources, secret_key)
     return token
+
+
+@v1.get("/restricted-queues")
+@v1.output(schemas.RestrictedQueueOut(many=True))
+def get_all_restricted_queues() -> list[dict]:
+    """List all agent's restricted queues and its owners."""
+    restricted_queues = database.get_restricted_queues()
+    restricted_queues_owners = database.get_restricted_queues_owners()
+
+    response = []
+    for queue in restricted_queues:
+        owners = restricted_queues_owners.get(queue, [])
+        response.append(
+            {
+                "queue": queue,
+                "owners": owners,
+            }
+        )
+
+    return jsonify(response)
+
+
+@v1.get("/restricted-queues/<queue_name>")
+@v1.output(schemas.RestrictedQueueOut)
+def get_restricted_queue(queue_name: str) -> dict:
+    """Get restricted queues for a specific agent."""
+    if not database.restricted_queue_exists(queue_name):
+        return jsonify(
+            {"error": "Restricted queue not found"}
+        ), HTTPStatus.NOT_FOUND
+
+    restricted_queues_owners = database.get_restricted_queues_owners()
+    owners = restricted_queues_owners.get(queue_name, [])
+
+    return jsonify(
+        {
+            "queue": queue_name,
+            "owners": owners,
+        }
+    )
+
+
+@v1.post("/restricted-queues/<queue_name>")
+@v1.input(schemas.RestrictedQueueIn, location="json")
+def add_restricted_queue(queue_name: str, json_data: dict) -> dict:
+    """Add an owner to the specific restricted queue."""
+    client_id = json_data.get("client_id", "")
+
+    if not client_id:
+        return jsonify({"error": "Missing client ID."}), HTTPStatus.BAD_REQUEST
+
+    if not database.queue_exists(queue_name):
+        return jsonify(
+            {"error": "No agent is associated with the specified queue."}
+        ), HTTPStatus.NOT_FOUND
+
+    database.add_restricted_queue(queue_name, client_id)
+
+    return "OK"
+
+
+@v1.delete("/restricted-queues/<queue_name>")
+@v1.input(schemas.RestrictedQueueIn, location="json")
+def delete_restricted_queue(queue_name: str, json_data: dict) -> dict:
+    """Delete an owner from the specific restricted queue."""
+    if not database.restricted_queue_exists(queue_name):
+        return jsonify(
+            {"error": "Restricted queue not found"}
+        ), HTTPStatus.NOT_FOUND
```

Comment:
For this, consider using apiflask abort instead:
```suggestion
        abort(HTTPStatus.NOT_FOUND, "Error: Restricted queue not found.")
```